File: backend/app/schemas/snapshot.py
from typing import Optional
from pydantic import BaseModel


class SnapshotCreate(BaseModel):
    '''
    - map_id: string
    - image_url: string
    - pose: dict
    - intrinsics: dict
    - meta: dict, default 값=None
    '''
    map_id: str
    image_url: str
    pose: dict
    intrinsics: dict
    meta: Optional[dict] = None


class SnapshotOut(BaseModel):
    '''
    - id: int
    - map_id: string
    - image_url: string
    - pose: dict
    - intrinsics: dict
    - meta: dict, default 값=None
    '''
    id: int
    map_id: str
    image_url: str
    pose: dict
    intrinsics: dict
    meta: Optional[dict] = None

File: frontend/src/types/snapshot.ts
export type Pose = {
  pos: number[];
  rot: number[];
};

export type Intrinsics = {
  fx: number;
  fy: number;
  cx: number;
  cy: number;
};

export type SnapshotPayload = {
  map_id: string;
  image_url: string;
  pose: Pose;
  intrinsics: Intrinsics;
  meta?: Record<string, unknown>;
};
